src/data/finset/basic.lean

```diff
@@ -2603,6 +2603,14 @@ begin
   { rw disjoint_iff, change _ ∩ _ = ∅, ext ⟨a, b⟩, rw mem_inter, finish, },
 end
 
+lemma finset.product_empty_left (t : finset β) :
+  (∅ : finset α).product t = ∅ :=
+rfl
+
+lemma finset.product_empty_right (s : finset α) :
```

Comment:
I think the names `empty_product` and `product_empty` would be more appropriate here.

Reply:
I'm fixed feeling about this convention, because it breaks precedence, but sure.

Reply:
There's precedence for @b-mehta's suggestion,`set.empty_prod` and `set.prod_empty`